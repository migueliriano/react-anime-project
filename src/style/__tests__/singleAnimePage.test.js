import React from 'react';
import renderer from 'react-test-renderer';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import toJson from 'enzyme-to-json';

import { shallow } from 'enzyme';

import {
  SinglePageContainer,
  CircularProgressStyle,
  CharacterBodyContainer,
  LogoLinkContainer,
  ColumnAnimeinfo,
  ColumnBodyPage,
} from '../singleAnimePage';

const muiTheme = getMuiTheme();
describe('singleAnimePage style components', () => {
  it('Should render a SinglePageContainer Component', () => {
    const tree = renderer.create(<SinglePageContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a CircularProgressStyle Component', () => {
    const tree = shallow(
      <CircularProgressStyle />,
      {
        context: muiTheme,
      },
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('Should render a CharacterBodyContainer Component', () => {
    const tree = renderer.create(<CharacterBodyContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a LogoLinkContainer Component', () => {
    const tree = renderer.create(<LogoLinkContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a ColumnAnimeinfo Component', () => {
    const tree = renderer.create(<ColumnAnimeinfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a ColumnBodyPage Component', () => {
    const tree = renderer.create(<ColumnBodyPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
