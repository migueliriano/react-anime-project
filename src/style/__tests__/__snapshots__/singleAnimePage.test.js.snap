// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`singleAnimePage style components Should render a CharacterBodyContainer Component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`singleAnimePage style components Should render a CircularProgressStyle Component 1`] = `
.c0 {
  margin: 0 auto;
}

<CircularProgress
  className="c0"
  max={100}
  min={0}
  mode="indeterminate"
  size={40}
  thickness={3.5}
  value={0}
/>
`;

exports[`singleAnimePage style components Should render a ColumnAnimeinfo Component 1`] = `
.c0 {
  margin-left: 15px;
  display: inline-block;
}

@media (max-width:768px) {
  .c0 {
    margin-left: 0;
    width: 100%;
    text-align: center;
    margin-bottom: 30px;
  }
}

<div
  className="c0"
/>
`;

exports[`singleAnimePage style components Should render a ColumnBodyPage Component 1`] = `
.c0 {
  margin-left: 15px;
  display: inline-block;
  margin-left: 50px;
  vertical-align: top;
}

@media (max-width:768px) {
  .c0 {
    margin-left: 0;
    width: 100%;
    text-align: center;
    margin-bottom: 30px;
  }
}

<div
  className="c0"
/>
`;

exports[`singleAnimePage style components Should render a LogoLinkContainer Component 1`] = `
.c0 {
  text-align: center;
  display: block;
}

<div
  className="c0"
/>
`;

exports[`singleAnimePage style components Should render a SinglePageContainer Component 1`] = `
<div
  className=""
/>
`;
